projects/nx-verdaccio/src/plugin/caching.unit-test.ts

```diff
@@ -0,0 +1,168 @@
+  });
+
+  describe('getCacheRecord', (): void => {
+    it('should call cacheKey once with correct arguments', (): void => {
+      getCacheRecord(targetsCache, prefix, hashData);
+      expect(cacheKeySpy).toHaveBeenCalledTimes(1);
+      expect(cacheKeySpy).toHaveBeenCalledWith(prefix, hashData);
+    });
+
+    it('should return the correct cache record if there is a cache hit', (): void => {
+      expect(getCacheRecord(targetsCache, prefix, hashData)).toEqual(cacheItem);
+    });
+
+    it('should return undefined if there is no cache hit', (): void => {
+      cacheKeySpy.mockReturnValue('non-existent-key');
+      expect(getCacheRecord(targetsCache, prefix, hashData)).toBeUndefined();
+    });
+  });
+
+  describe('setCacheRecord', (): void => {
+    const cacheData = { thunderfury: 'Blood of Sylvanas' };
+
+    it('should call cacheKey once with correct arguments', (): void => {
+      setCacheRecord(targetsCache, prefix, hashData, cacheData);
+      expect(cacheKeySpy).toHaveBeenCalledTimes(1);
+      expect(cacheKeySpy).toHaveBeenCalledWith(prefix, hashData);
+    });
+
+    it('should set the cache record, and return it', (): void => {
+      expect(setCacheRecord(targetsCache, prefix, hashData, cacheData)).toBe(
+        cacheData
+      );
+      expect(targetsCache).toHaveProperty(cacheKey, cacheData);
+    });
+
+    it('should update existing cache data, and return it', (): void => {
+      const recordToUpdate = { thunderfury: 'Soul of Sylvanas' };
+      setCacheRecord(targetsCache, prefix, hashData, cacheData);
+
+      expect(
+        setCacheRecord(targetsCache, prefix, hashData, recordToUpdate)
+      ).toBe(recordToUpdate);
+      expect(targetsCache).toHaveProperty(cacheKey, recordToUpdate);
+    });
+  });
+});
+
+describe('readTargetsCache', (): void => {
+  const path = 'azeroth';
+  let existsSyncSpy: MockInstance<[path: nodeFs.PathLike], boolean>;
+  let readJsonFileSpy: MockInstance<
+    [path: string, options?: JsonReadOptions],
+    object
+  >;
+
+  beforeEach((): void => {
+    existsSyncSpy = vi
+      .spyOn(nodeFs, 'existsSync')
+      .mockImplementation((): boolean => true);
+    readJsonFileSpy = vi
+      .spyOn(nxDevKit, 'readJsonFile')
+      .mockImplementation((): Record<string, Partial<ProjectConfiguration>> => {
+        return MOCK_TARGETS_CACHE;
+      });
+    vi.stubEnv("NX_CACHE_PROJECT_GRAPH", 'true')
+  });
+
+  afterEach((): void => {
+    existsSyncSpy.mockRestore();
+    readJsonFileSpy.mockRestore();
+    vi.clearAllMocks();
+  });
+
+  it('should call existSync once with correct argument', (): void => {
+    readTargetsCache(path);
+    expect(existsSyncSpy).toHaveBeenCalledWith(path);
+    expect(existsSyncSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call readJsonFile once with correct argument', (): void => {
+    readTargetsCache(path);
+    expect(readJsonFileSpy).toHaveBeenCalledWith(path);
+    expect(readJsonFileSpy).toHaveBeenCalledTimes(1);
```

Comment:
I would revert the order here. The number of calls is an insurance to avoid fals positives in case the method is called more then 1 time. the order let tis error pop up earlier.